kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -451,13 +452,19 @@ private long doCommit(
                     dataAndMetadataActions.map(
                         action -> {
                           transactionMetrics.totalActionsCounter.increment();
-                          // TODO: handle RemoveFiles.
                           if (!action.isNullAt(ADD_FILE_ORDINAL)) {
                             transactionMetrics.addFilesCounter.increment();
                             transactionMetrics.addFilesSizeInBytesCounter.increment(
                                 new AddFile(action.getStruct(ADD_FILE_ORDINAL)).getSize());
                           } else if (!action.isNullAt(REMOVE_FILE_ORDINAL)) {
                             transactionMetrics.removeFilesCounter.increment();
+                            // TODO add removeFileSizeInBytes and increment
+                            // TODO update fileSizeHistogram
```

Comment:
I will do this in a follow-up PR

Reply:
I would suggest doing it before this PR if you feel like some user would start using it right away. 

Before removeFileSizeInBytes in metric, CRC's `tableSizeInByte` will just be incorrect and hard to fix(because it actually materialize a actual file).

If you expect no immediate usage, then ordering should be fine.

Reply:
I think this will be fine to do in the next few days as a follow up. I don't expect this to be an issue